LiveStream/Views/Controllers/LiveVideoViewController.swift

```diff
@@ -78,6 +79,47 @@ public final class LiveVideoViewController: UIViewController {
         guard let _self = self else { return }
         self?.delegate?.liveVideoViewControllerPlaybackStateChanged(controller: _self, state: $0)
     }
+
+    self.viewModel.outputs.unsubscribeAllSubscribersFromSession
+      .observeForUI()
+      .observeValues { [weak self] in
+        guard let _self = self else { return }
+        _self.subscribers.forEach { subscriber in
+          _self.session?.unsubscribe(subscriber, error: nil)
+        }
+    }
+
+    self.viewModel.outputs.resubscribeAllSubscribersToSession
+      .observeForUI()
+      .observeValues { [weak self] in
+        guard let _self = self else { return }
+        _self.subscribers.forEach { subscriber in
+          _self.session?.subscribe(subscriber, error: nil)
+        }
+    }
+
+    self.viewModel.outputs.toggleHlsPause
+      .observeForUI()
+      .observeValues { [weak self] pause in
+        guard let _self = self else { return }
```

Comment:
i think we prefer `_self` if we're using animation closures or when passing data in a function that doesn't except nil parameters (such as tableview methods).